### PR TITLE
feat: Add `--bag` option to hold profits

### PR DESCRIPTION
**Pull request by jlaw**
Instead of selling all of a filled order to make profit,
sell only enough to break even and leave the profit in "free coins"

### CODE REVIEW

**Comment by svanas** [comment]
I struggle with the name. In my reptile brain, `--bag` sounds like a negative. Nobody will want to collect a bag of shitcoins. How about `--break-even` or `--earn-free-coins`?

File: exchanges/bittrex.go [hunk 1]
```diff
@@ -514,7 +514,14 @@ func (self *Bittrex) sell(
 					if err == nil {
 						var prec int
 						if prec, err = self.GetPricePrec(client, order.MarketName()); err == nil {
-							qty := self.GetMaxSize(client, base, quote, hold.HasMarket(order.MarketName()), order.QuantityFilled())
+							amount := order.QuantityFilled()
+							// only sell to break even and bag the remaining amount
+							if flag.Exists("bag") {
+								sp,_ := self.GetSizePrec(client, order.MarketName())
```
Comment (svanas) [inline exchanges/bittrex.go:520]: Please handle the error here (if any)

File: exchanges/gdax.go [hunk 2]
```diff
@@ -465,6 +465,13 @@ func (self *Gdax) sell(
 							}
 
 							qty := old.Size
+							// only sell to break even and bag the remaining amount
+							if flag.Exists("bag") {
+								sp,_ := self.GetSizePrec(client, msg.ProductId)
```
Comment (svanas) [inline exchanges/gdax.go:470]: Please see my earlier comment about error handling. We should never ignore them.

**Review by svanas — COMMENTED** [review]
Before we can merge this PR, we need an implementation for the following exchanges:
1. Bitstamp
2. HitBTC
3. Binance
4. crypto.com

It would be awesome if you could reuse this code rather than duplicate it, for example:
1. copy this code into `exchanges.GetMaxSize`, or
2. copy this code into `model.GetSizeMax`, or
3. introduce a new higher-level function in the `pricing` module .

**Comment by d00vy** [comment]
> I struggle with the name. In my reptile brain, `--bag` sounds like a negative. Nobody will want to collect a bag of shitcoins. How about `--break-even` or `--earn-free-coins`?

Maybe use `--build` since it's an account building strategy

**Comment by hidefsooner** [comment]
> > I struggle with the name. In my reptile brain, `--bag` sounds like a negative. Nobody will want to collect a bag of shitcoins. How about `--break-even` or `--earn-free-coins`?
> 
> Maybe use `--build` since it's an account building strategy

Or how about --account-building?

I would really love this feature. This is how I trade manually.

**Comment by svanas** [comment]
Implemented in 0.0.174 across all exchanges as --earn

**Pull request closed without merging**